### ISSUE TITLE
Exploring Face Detection Models

### ISSUE THREAD

**Issue by mayank8200** [akshitagupta15june/Face-X#89]
Face detection is the first step you need to work on.
So..The first task for you all will be exploring different algorithms that are used for face detection...
We can use the pre-trained face detection model.

**Comment by keshavbansal015** [comment]
Hey, @mayank8200 could you please elaborate on this issue?

**Comment by mayank8200** [comment]
Yes..If you look for face detection models over the Internet a bunch of models has been already trained. You just need to go through some of popular pretrained models and find out which one has superior accuracy among all.
If you want any resources then let me know. I will try to provide you.

**Comment by keshavbansal015** [comment]
Sure, do share some resources from where I can start exploring and learning. Also thank you for a prompt response!!

**Comment by akshitagupta15june** [comment]
@keshavbansal015 Would you like to work in this issue so i will assign you under swoc?

**Comment by keshavbansal015** [comment]
Sure, I can first try adding popular algorithms or models for face detection, and then move on to creating modules to compare the performances of different methods. I might need some help though.

**Comment by akshitagupta15june** [comment]
most of recognition algos are available in repo so have a look at them too

**Comment by keshavbansal015** [comment]
Yes, I can work models that are not currently available in the repo.

**Comment by srimontidutta** [comment]
Hello, can I please contribute to this issue for GSSoC 21?

**Comment by akshitagupta15june** [comment]
Yes @mayank8200 will be your Mentor

**Comment by srimontidutta** [comment]
Thank you!@akshitagupta15june

**Comment by mayank8200** [comment]
Face detection is the first step you need to work on.
So...The first task for you all will be exploring different algorithms that are used for face detection...
We can use the pre-trained face detection model.
 If you look for face detection models over the Internet a bunch of models has been already trained. You just need to go through some of the popular pre-trained models and find out which one has superior accuracy among all.

**Comment by Hamza3226** [comment]
Hello, I am a GSSOC'21 participant and would like to contribute to this project.

**Comment by akshitagupta15june** [comment]
hi @Hamza3226 your mentor will be @mayank8200

**Comment by Hamza3226** [comment]
> hi @Hamza3226 your mentor will be @mayank8200

Ok thank-you ma'am

**Comment by Sibasish-Padhy** [comment]
hello,I am a GSSOC '21 participant and would like to contribute to this project

**Comment by jeelnathani** [comment]
hello. I am GSSOC'21 participant and I would like contribute here on nose detection.

**Comment by Hamza3226** [comment]
> hello. I am GSSOC'21 participant and I would like contribute here on nose detection.

Sorry, but this project is already assigned to me.

**Comment by jeelnathani** [comment]
> > hello. I am GSSOC'21 participant and I would like contribute here on nose detection.
> 
> Sorry, but this project is already assigned to me.



> Hello, I am a GSSOC'21 participant and would like to contribute to this project.

@Hamza3226 you haven't explicitly mentioned on which module you want to contribute, which created ambiguity here

**Comment by Hamza3226** [comment]
> > > hello. I am GSSOC'21 participant and I would like contribute here on nose detection.
> > 
> > 
> > Sorry, but this project is already assigned to me.
> 
> > Hello, I am a GSSOC'21 participant and would like to contribute to this project.
> 
> @Hamza3226 you haven't explicitly mentioned on which module you want to contribute, which created ambiguity here

I researched about various face detection models and found out the most efficient one. Currently i am researching about transfer Learning.

**Comment by jeelnathani** [comment]
> > > > hello. I am GSSOC'21 participant and I would like contribute here on nose detection.
> > > 
> > > 
> > > Sorry, but this project is already assigned to me.
> > 
> > 
> > > Hello, I am a GSSOC'21 participant and would like to contribute to this project.
> > 
> > 
> > @Hamza3226 you haven't explicitly mentioned on which module you want to contribute, which created ambiguity here
> 
> I researched about various face detection models and found out the most efficient one. Currently i am researching about transfer Learning

are you working on face detection only!

**Comment by Hamza3226** [comment]
> > > > > hello. I am GSSOC'21 participant and I would like contribute here on nose detection.
> > > > 
> > > > 
> > > > Sorry, but this project is already assigned to me.
> > > 
> > > 
> > > > Hello, I am a GSSOC'21 participant and would like to contribute to this project.
> > > 
> > > 
> > > @Hamza3226 you haven't explicitly mentioned on which module you want to contribute, which created ambiguity here
> > 
> > 
> > I researched about various face detection models and found out the most efficient one. Currently i am researching about transfer Learning
> 
> are you working on face detection only!

No, also working on snapchat filters also

**Comment by akshitagupta15june** [comment]
See haarcascade is implemented now so @mayank8200 please assign each participants different models

**Comment by akshitagupta15june** [comment]
this issue is still opened so participants can work on it!

**Comment by subhangi2731** [comment]
@akshitagupta15june please assign me this issue

**Comment by akshitagupta15june** [comment]
ok assigning @subhangi2731

**Comment by kuluruvineeth** [comment]
@akshitagupta15june i would like to enhance code part and readme part for face detection using MTCNN pictorially and pipeline for that algorithm.
so please assign the issue #89 to explore and enhance the face detection model.
Hope you shall do the needful,expeditiously.
Thank you
vineeth

**Comment by akshitagupta15june** [comment]
ok @kuluruvineeth are a lgm-soc participant?

**Comment by kuluruvineeth** [comment]
No @akshitagupta15june  but super passionate and have deep insight and knowledge about deep learning for computer vision...If I am not eligible to contribute I would be happy to know other options from your kind self and that you would be kindness of you.
Thank you 
Vineeth

**Comment by akshitagupta15june** [comment]
It's ok i was just asking as many participants are from different open source program so we have to allot labels in them.
Open source is for everyone, you can contribute in this repo.
Let me know what do you want to implement by creating a new issue

**Comment by 1anshu-56** [comment]
I want to work on this issue as i have experience in OpenCV can you please assign it to me.

**Comment by SURABHI-GUPTA** [comment]
I would like to contribute to this thread.

**Comment by akshitagupta15june** [comment]
Ok i am assigning you

**Comment by C-Omkar** [comment]
I would like to contribute to this thread.

**Comment by akshitagupta15june** [comment]
You can work on it look at above comments for more info

**Comment by hemhemoh** [comment]
Hi, I am interested in this issue. What face detection model do you think I should explore? VGGFace? Resnet?

**Comment by hemhemoh** [comment]
okay, @akshitagupta15june Just to let you know I have started working on the issues assigned to me. Sorry for the delay, was ill

**Comment by anusha-shanbhog** [comment]
Hey I am interested to work on this for KWOC'22. Could you guide me as to what exactly I need to do?

**Comment by akshitagupta15june** [comment]
Please look at above comments

**Comment by PraNavKumAr01** [comment]
Ive been learning and building face detection, expression detection and material detection on the faces. I would like to work on this issue as a part of SSOC'23
Please assign me this issue

**Comment by ayush-09** [comment]
Hey @akshitagupta15june
I am interested in working on issue #89, "Exploring Face Detection Models." I have experience with face detection models, including haarcascade, mediapipe, and CNN. I would like to contribute to this issue by exploring different approaches and techniques to improve the accuracy of face detection.
I would appreciate it if you could assign me this issue. I am excited about the opportunity to contribute to the project and collaborate with the team.
Thank you

**Comment by ayush-09** [comment]
Hey @akshitagupta15june
I am interested in working on issue "Exploring Face Detection Models." I have experience with face detection models, including haarcascade, mediapipe, and CNN. I would like to contribute to this issue by exploring different approaches and techniques to improve the accuracy of face detection.
I would appreciate it if you could assign me this issue. I am excited about the opportunity to contribute to the project and collaborate with the team.
Thank you
